flink-runtime/src/test/java/org/apache/flink/runtime/util/profiler/ProfilingServiceTest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util.profiler;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.RestOptions;
+import org.apache.flink.runtime.rest.messages.ProfilingInfo;
+import org.apache.flink.util.StringUtils;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/** Unit tests for {@link ProfilingService}. */
+public class ProfilingServiceTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProfilingServiceTest.class);
+    private static final Configuration configs = new Configuration();
+    private static final String NO_ACCESS_TO_PERF_EVENTS = "No access to perf events.";
+    private static final String NO_ALLOC_SYMBOL_FOUND = "No AllocTracer symbols found.";
+    private static final String resourceID = "TestJobManager";
+    private static final long profilingDuration = 3L;
+    private static final int historySizeLimit = 3;
+
+    private ProfilingService profilingService;
+
+    @BeforeAll
+    static void beforeAll() {
+        configs.set(RestOptions.MAX_PROFILING_HISTORY_SIZE, historySizeLimit);
+    }
+
+    @BeforeEach
+    void setUp(@TempDir Path tempDir) {
+        configs.set(RestOptions.PROFILING_RESULT_DIR, tempDir.toString());
+        profilingService = ProfilingService.getInstance(configs);
+        verifyConfigsWorks(profilingService, tempDir);
```

Comment:
I don't think we need to verify the configs within the `setup`.

Reply:
Since we initialized `ProfilingService` with `tempDir` created by junit here (generated in runtime, not accessible in `@Test`), so I checked the configuration of the profiling result directory works within the `setUp` process.

Reply:
The `setUp` method run `@BeforeEach` should not include any test behavior. If you want to add a test to verify the configs, please add a separate unit test to verify it.

For the `tmpeDir` problem, you can introduce `@TempDir Path tempDir` as a private field in the `ProfilingServiceTest` class to avoid such a varible in the `setUp` method.

Reply:
I have extracted a single test for checking the configuration works, which was made by comparing the configs in the `ProfilingServiceTest` and variables in `ProfilingService`.